Bottom: 6ce21e3ce6012d1136f69d476e10f4b4a1195da9
Top:    82701973de20fd9ef1198cbcf8b704c0b18cce8e
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-12-16 15:38:27 +0800

Refresh of macvtap-xdp-fast-rx-support

---

diff --git a/drivers/net/macvtap.c b/drivers/net/macvtap.c
index 5da9861..5b4654a 100644
--- a/drivers/net/macvtap.c
+++ b/drivers/net/macvtap.c
@@ -45,6 +45,7 @@ struct macvtap_queue {
 	bool enabled;
 	struct list_head next;
 	struct skb_array skb_array;
+	struct xdp_array xdp_array;
 };
 
 #define MACVTAP_FEATURES (IFF_VNET_HDR | IFF_MULTI_QUEUE)
@@ -580,6 +581,9 @@ static int macvtap_open(struct inode *inode, struct file *file)
 	if (skb_array_init(&q->skb_array, dev->tx_queue_len, GFP_KERNEL))
 		goto err_array;
 
+	if (ptr_ring_init(&q->xdp_array, dev->xdp_array, GFP_KERNEL))
+		goto err_xdp;
+
 	err = macvtap_set_queue(dev, file, q);
 	if (err)
 		goto err_queue;
@@ -590,6 +594,8 @@ static int macvtap_open(struct inode *inode, struct file *file)
 	return err;
 
 err_queue:
+	ptr_ring_cleanup(&q->xdp_array, kfree);
+err_xdp:
 	skb_array_cleanup(&q->skb_array);
 err_array:
 	sock_put(&q->sk);
